src/Registry/Registry.spec.ts

```diff
@@ -75,16 +75,16 @@ describe( "Registry", () => {
 				} ).toThrowError( IllegalArgumentError, `"id" is out of scope.` );
 			} );
 
-			it( "should throw error when ID already taken", () => {
+			it( "should return resource", () => {
 				const registry:Registry<RegisteredPointer> = createMock( {} );
+
 				registry.__resourcesMap.set( "id", RegisteredPointer.create( { $registry: registry } ) );
 
 				expect( () => {
 					registry._addPointer( { $id: "id" } );
-				} ).toThrowError( IDAlreadyInUseError, `"id" is already being used.` );
+				} ).toEqual( jasmine.any( Function ) );
```

Comment:
For each test we should review if the returned value equals the resource first set.